compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -176,21 +176,36 @@ uint8_t getMemoryBarrierBinaryLengthLowerBound(int32_t barrier, TR::CodeGenerato
 
 // -----------------------------------------------------------------------------
 // OMR::X86::Instruction:: member functions
+bool OMR::X86::Instruction::needsLockPrefix()
+   {
+   return self()->getOpCode().needsLockPrefix();
+   }
 
-uint8_t *OMR::X86::Instruction::generateBinaryEncoding()
+uint8_t* OMR::X86::Instruction::generateBinaryEncoding()
    {
-   // *this    swipeable for debugging purposes
-   uint8_t *instructionStart = self()->cg()->getBinaryBufferCursor();
-   uint8_t *cursor           = instructionStart;
-   if (self()->getOpCode().needsRepPrefix())
+   for(;;)
```

Comment:
We normally exit the loop within one iteration. The loop is to handle the case operand() requests regenerating current instruction by returning NULL. The previous approach was to recursive call itself, this proposed approach uses a loop to be safer and more efficient.